gradle-plugin/configure/build.gradle.kts

```diff
@@ -1,28 +1,28 @@
-import com.novoda.gradle.release.PublishExtension
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 /// The plugin version number – change this to match whatever your tag will be
-version = "0.5.0"
 group = "com.automattic.android"
+version = "0.6.0"
 
 buildscript {
     repositories {
-        jcenter()
-        maven {
-            setUrl("https://plugins.gradle.org/m2/")
-        }
+        maven { url = uri("https://a8c-libs.s3.amazonaws.com/android") }
     }
     dependencies {
-        classpath("com.novoda", "bintray-release", "0.9.2")
+        classpath("com.automattic.android:publish-to-s3:0.3.1")
     }
 }
 
 plugins {
     `kotlin-dsl`
-    id("com.github.gmazzo.buildconfig") version "2.0.2"
+    id("com.github.gmazzo.buildconfig") version "3.0.0"
 }
 
-apply(null, "com.novoda.bintray-release")
+apply(plugin = "com.automattic.android.publish-plugin-to-s3")
+configure<com.automattic.android.publish.PublishPluginToS3Extension> {
```

Comment:
[Trying to improve my own knowledge again]
What is this in brackets supposed to be? I see it's different to the `com.automattic.android.publish-plugin-to-s3` that I assumed was kind of the way to identify the publish-to-s3 plugin, so where does `com.automattic.android.publish.PublishPluginToS3Extension` come from and why can't we use the `com.automattic.android.publish-plugin-to-s3` identifier there instead?
(personally feels like it makes it hard to connect the dots and realize that they are related to the same thing – luckily despite the difference in case and structure of that identifier, they use similar words, but I can't imagine if whatever that identifier is was way more different than the other 🤔 )

Reply:
So... Kotlin DSL, in some cases, is quite different from its Groovy counterpart, and unfortunately neither syntax carry enough information about what's going on, so here are a bunch of pieces so you can connect the dots:
* Gradle plugins can have extensions, which is basically [a way for clients to configure the behavior of the plugin](https://docs.gradle.org/current/userguide/custom_plugins.html#sec:getting_input_from_the_build).
* In publish-to-s3-gradle-plugin project, we have a [PublishPluginToS3Extension](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/src/main/kotlin/com/automattic/android/PublishPluginToS3Extension.kt) which is [created here](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/src/main/kotlin/com/automattic/android/PublishPluginToS3Plugin.kt#L13).
* Groovy syntax will use the name of the extension, instead of its type, so in this case it'd have been `s3PublishPlugin` because [that's what the name we give it here](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/src/main/kotlin/com/automattic/android/PublishPluginToS3Plugin.kt#L13). An example of this is [where we set the extension properties with `s3PublishLibrary` block in WordPress-Utils-Android](https://github.com/wordpress-mobile/WordPress-Utils-Android/blob/develop/WordPressUtils/build.gradle#L90-L94) which is [defined here](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/src/main/kotlin/com/automattic/android/PublishLibraryToS3Extension.kt) and [created here](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/src/main/kotlin/com/automattic/android/PublishLibraryToS3Plugin.kt#L14).

What all this also means is that, when we see an extension usage in a build file, we don't really know where it comes from without doing some research.